### PR TITLE
document more about `proxy_url` and `get_updates_proxy_url`

### PR DESCRIPTION
**Pull request by trim21**

### CODE REVIEW

File: telegram/ext/_applicationbuilder.py [hunk 1]
```diff
@@ -640,6 +645,10 @@ def get_updates_proxy_url(self: BuilderType, get_updates_proxy_url: str) -> Buil
         """Sets the proxy for the :paramref:`telegram.request.HTTPXRequest.proxy_url`
         parameter which is used for :meth:`telegram.Bot.get_updates`. Defaults to :obj:`None`.
 
+        proxy used to get updates.
```
Comment (harshil21) [inline telegram/ext/_applicationbuilder.py:648]:
```suggestion
```

already explained by this method's docstring.

File: telegram/ext/_applicationbuilder.py [hunk 2]
```diff
@@ -487,6 +487,11 @@ def proxy_url(self: BuilderType, proxy_url: str) -> BuilderType:
         """Sets the proxy for the :paramref:`~telegram.request.HTTPXRequest.proxy_url`
         parameter of :attr:`telegram.Bot.request`. Defaults to :obj:`None`.
 
+        Note:
+            this is not the proxy used for all http requests.
```
Comment (harshil21) [inline telegram/ext/_applicationbuilder.py:491]:
```suggestion
```
I don't think this note is necessary because of the 'see also' you added below

**Review by harshil21 — CHANGES_REQUESTED** [review]
Thanks for the PR. I would be okay with adding this 'see also' note in all the `(get_updates_)*` methods in AppBuilder. What say others?

File: telegram/ext/_applicationbuilder.py (hunk 2 above)
Comment (trim21) [reply telegram/ext/_applicationbuilder.py:491]: As a user personally I think it's necessary to note this...

**Comment by Bibo-Joshi** [comment]
I would prefer to just add a `.. seealso:: get_updates_proxy_url` (+ vice versa + for all other request related methods that have two variants)

File: telegram/ext/_applicationbuilder.py (hunk 2 above)
Comment (harshil21) [reply telegram/ext/_applicationbuilder.py:491]: hinrich also voted to remove this note (https://github.com/python-telegram-bot/python-telegram-bot/pull/3816#issuecomment-1654517762)

**Review by Bibo-Joshi — COMMENTED** [review]
Thanks for the updates! Would you mind also adding similar changes to the other `get_updates_*` methods and their counterparts?

**Comment by trim21** [comment]
> Thanks for the updates! Would you mind also adding similar changes to the other `get_updates_*` methods and their counterparts?

I do mind...

**Comment by Bibo-Joshi** [comment]
Fair enough, thanks for the contribution in any case!

**Pull request merged**